assets/controllers/password_generator_controller.js

```diff
@@ -0,0 +1,24 @@
+import {Controller} from '@hotwired/stimulus'
+
+/* stimulusFetch: 'lazy' */
+export default class extends Controller {
+  static targets = ['input'];
+
+  generate() {
+    this.inputTarget.value = this.getGeneratedPassword(8);
+  }
+
+  getGeneratedPassword(length) {
+    let result = '';
+    const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+    const charactersLength = characters.length;
+    let i = 0;
+
+    while (i < length) {
+      result += characters.charAt(Math.floor(Math.random() * charactersLength));
+      i++;
+    }
+
+    return result;
```

Comment:
you could be doing something more functionnal style like

```es6
return (new Array(characters.length))
  .reduce((acc) => acc + characters.charAt(Math.floor(Math.random() * characters.length)), '');
```

or even mofe a bit of code in a function
```
const getRandomByte() => this.characters.length.charAt(Math.floor(Math.random() * this.characters.length));

...

return (new Array(characters.length))
  .reduce((acc) => acc + this->getRandomByte(), '');
```